File: 프로그래머스/0/181887. 홀수 vs 짝수/홀수 vs 짝수.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> num_list) {
    int answer = 0, oddSum = 0, evenSum = 0;
    int count = 1;
    
    for(const auto& value : num_list) {
        (count % 2 != 0) ? oddSum += value : evenSum += value;
        count++;
    }
    
    if(oddSum >= evenSum) {
        return oddSum;
    } else {
        return evenSum;
    }
    
}

File: 프로그래머스/0/181887. 홀수 vs 짝수/홀수 vs 짝수.swift
import Foundation

func solution(_ num_list:[Int]) -> Int {
    
    var evenSum = 0
    var oddSum = 0
    
    num_list.enumerated().forEach { ($0+1).isMultiple(of: 2) ? (evenSum += $1) : (oddSum += $1) }
    
    return (evenSum > oddSum) ? evenSum : oddSum
}
